### PR TITLE
Squeezebox plugins: Migrate to raw socket connection

### PR DESCRIPTION
**Pull request by luk1337**
Check-list
----------

 * [x] There is a linked issue discussing the motivations for this feature or bugfix
 * [ ] Unit tests have been added where possible
 * [ ] I've added / updated documentation for any user-facing features.
 * [x] Performance seems to be comparable or better than current `main`

What this change is adding / fixing
-----------------------------------
Python 3.13 removed telnetlib, so we ought to migrate away from it, see #4473.

### CODE REVIEW

**Comment by declension** [comment]
Thanks. There are no unit tests, and I no longer have access to a system to test with. Can you confirm manual testing works?

**Comment by luk1337** [comment]
> Thanks. There are no unit tests, and I no longer have access to a system to test with. Can you confirm manual testing works?

I explained my testing in the commit msg https://github.com/quodlibet/quodlibet/pull/4563/commits/f2fa1f37b164507015adfb863bf2e53f81f3e67e. I couldn't test more than that, cause I had 0 players added.

File: quodlibet/ext/_shared/squeezebox/server.py [hunk 1]
```diff
@@ -99,10 +95,10 @@ def __request(self, line, raw=False, want_reply=True):
         if self._debug:
             print_(f'>>>> "{line}"')
         try:
-            self.telnet.write((line + "\n").encode("utf-8"))
+            self.socket.send((line + "\n").encode("utf-8"))
             if not want_reply:
                 return None
-            raw_response = self.telnet.read_until(b"\n", 5).decode("utf-8")
+            raw_response = self.socket.recv(1024).decode("utf-8")
```
Comment (declension) [inline quodlibet/ext/_shared/squeezebox/server.py:101]: This no longer checks for the newline though; I fear if we've pipelined several responses back we'll accidentally parse both. Maybe that's never an issue in real life though.

(also 1024 feels a little small, though I can't remember how big they get here)

Comment (luk1337) [reply quodlibet/ext/_shared/squeezebox/server.py:101]: I could increase it to 4k, but it seems like 1024 might be enough, as I get responses like:

```
00%3Ad8%3A61%3Ac8%3Aae%3Add playlist play %2Fmnt%2Fdata%2FMusic%2FMonogatari%20Series%20%C3%A7%C2%89%C2%A9%C3%A8%C2%AA%C2%9E%C3%A3%C2%82%C2%B7%C3%A3%C2%83%C2%AA%C3%A3%C2%83%C2%BC%C3%A3%C2%82%C2%BA%2FMonogatari%20Series%20%C3%A7%C2%89%C2%A9%C3%A8%C2%AA%C2%9E%C3%A3%C2%82%C2%B7%C3%A3%C2%83%C2%AA%C3%A3%C2%83%C2%BC%C3%A3%C2%82%C2%BA%20UTAMONOGATARI%2F%5B2019.12.25%5D%20%C3%A7%C2%89%C2%A9%C3%A8%C2%AA%C2%9E%C3%A3%C2%82%C2%B7%C3%A3%C2%83%C2%AA%C3%A3%C2%83%C2%BC%C3%A3%C2%82%C2%BA%C3%AF%C2%BD%C2%9CUTAMONOGATARI%20-DISK%201-%2F04.%20%C3%A6%C2%81%C2%8B%C3%A6%C2%84%C2%9B%C3%A3%C2%82%C2%B5%C3%A3%C2%83%C2%BC%C3%A3%C2%82%C2%AD%C3%A3%C2%83%C2%A5%C3%A3%C2%83%C2%AC%C3%A3%C2%83%C2%BC%C3%A3%C2%82%C2%B7%C3%A3%C2%83%C2%A7%C3%A3%C2%83%C2%B3.ogg
```

Also, if you'd like to try it out, `docker run --net=host -v $PWD:/mnt/music:Z justifiably/logitechmediaserver` plus `squeezelite` appears to work fine with this PR.

Comment (luk1337) [reply quodlibet/ext/_shared/squeezebox/server.py:101]: Also, if you're afraid of getting multiple responses, I could do ` raw_response = self.socket.recv(1024).decode("utf-8").split("\n")[0]`

Comment (luk1337) [reply quodlibet/ext/_shared/squeezebox/server.py]: I switched it to recv until newline, is this good enough now?

**Pull request merged**